bin/utils/cleanup_and_exit

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+#                         License
+#
+# Copyright (C) 2024  David Valin dvalin@redhat.com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# Report and record any passed message and perform any of the following,
+# depending on the passed arguments.
+#
+# 1) Delete the lock file
+# 2) Restore a set of files.
+# 3) Remove the cli_supplied options file.
+#
+# At the end we will exit out with the return code passed.  If no return
+# code is passed, then a value of 0 is returned by default.
+#
+# Note this script should be called via exec, that way when we exit, we will exit
+# back to the shell, not the calling program.
+#
```

Comment:
Instead of requiring an `exec` maybe we should refactor this so we `source` the exit code into the original script, and then call exit.

This way we don't need to do anything special to ensure that the script dies when it should.